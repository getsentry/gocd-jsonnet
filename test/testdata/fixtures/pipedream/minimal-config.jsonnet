local pipedream = import '../../../../libs/pipedream.libsonnet';

local pipedream_config = {
  name: 'example',
  materials: {
    init_repo: {
      git: 'git@github.com:getsentry/init.git',
      branch: 'master',
      destination: 'init',
    },
  },
};

local sample = {
  pipeline(region):: {
    region: region,
    materials: {
      example_repo: {
        git: 'git@github.com:getsentry/example.git',
        shallow_clone: true,
        branch: 'master',
        destination: 'example',
      },
    },
    stages: [
      {
        example_stage: {},
      },
    ],
  },
};

pipedream.render(pipedream_config, sample.pipeline)
