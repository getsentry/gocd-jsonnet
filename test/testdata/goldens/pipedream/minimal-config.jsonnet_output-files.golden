{
   "deploy-example-customer-1.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example-customer-1": {
            "display_order": 5,
            "group": "example",
            "materials": {
               "deploy-example-de-pipeline-complete": {
                  "pipeline": "deploy-example-de",
                  "stage": "pipeline-complete"
               },
               "example_repo": {
                  "branch": "master",
                  "destination": "example",
                  "git": "git@github.com:getsentry/example.git",
                  "shallow_clone": true
               }
            },
            "region": "customer-1",
            "stages": [
               {
                  "example_stage": { }
               },
               {
                  "pipeline-complete": {
                     "jobs": {
                        "pipeline-complete": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   },
   "deploy-example-customer-2.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example-customer-2": {
            "display_order": 6,
            "group": "example",
            "materials": {
               "deploy-example-customer-1-pipeline-complete": {
                  "pipeline": "deploy-example-customer-1",
                  "stage": "pipeline-complete"
               },
               "example_repo": {
                  "branch": "master",
                  "destination": "example",
                  "git": "git@github.com:getsentry/example.git",
                  "shallow_clone": true
               }
            },
            "region": "customer-2",
            "stages": [
               {
                  "example_stage": { }
               },
               {
                  "pipeline-complete": {
                     "jobs": {
                        "pipeline-complete": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   },
   "deploy-example-customer-4.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example-customer-4": {
            "display_order": 8,
            "group": "example",
            "materials": {
               "deploy-example-customer-2-pipeline-complete": {
                  "pipeline": "deploy-example-customer-2",
                  "stage": "pipeline-complete"
               },
               "example_repo": {
                  "branch": "master",
                  "destination": "example",
                  "git": "git@github.com:getsentry/example.git",
                  "shallow_clone": true
               }
            },
            "region": "customer-4",
            "stages": [
               {
                  "example_stage": { }
               },
               {
                  "pipeline-complete": {
                     "jobs": {
                        "pipeline-complete": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   },
   "deploy-example-de.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example-de": {
            "display_order": 4,
            "group": "example",
            "materials": {
               "deploy-example-us-pipeline-complete": {
                  "pipeline": "deploy-example-us",
                  "stage": "pipeline-complete"
               },
               "example_repo": {
                  "branch": "master",
                  "destination": "example",
                  "git": "git@github.com:getsentry/example.git",
                  "shallow_clone": true
               }
            },
            "region": "de",
            "stages": [
               {
                  "example_stage": { }
               },
               {
                  "pipeline-complete": {
                     "jobs": {
                        "pipeline-complete": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   },
   "deploy-example-s4s.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example-s4s": {
            "display_order": 2,
            "group": "example",
            "materials": {
               "example_repo": {
                  "branch": "master",
                  "destination": "example",
                  "git": "git@github.com:getsentry/example.git",
                  "shallow_clone": true
               }
            },
            "region": "s4s",
            "stages": [
               {
                  "example_stage": { }
               },
               {
                  "pipeline-complete": {
                     "jobs": {
                        "pipeline-complete": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   },
   "deploy-example-us.yaml": {
      "format_version": 10,
      "pipelines": {
         "deploy-example-us": {
            "display_order": 3,
            "group": "example",
            "materials": {
               "deploy-example-s4s-pipeline-complete": {
                  "pipeline": "deploy-example-s4s",
                  "stage": "pipeline-complete"
               },
               "example_repo": {
                  "branch": "master",
                  "destination": "example",
                  "git": "git@github.com:getsentry/example.git",
                  "shallow_clone": true
               }
            },
            "region": "us",
            "stages": [
               {
                  "example_stage": { }
               },
               {
                  "pipeline-complete": {
                     "jobs": {
                        "pipeline-complete": {
                           "tasks": [
                              {
                                 "exec": {
                                    "command": true
                                 }
                              }
                           ]
                        }
                     }
                  }
               }
            ]
         }
      }
   }
}
